### ISSUE TITLE
How to play back XDF file into outlet(s)

### ISSUE THREAD

**Issue by cboulay** [sccn/labstreaminglayer#93]
I'd like to play back some old data files for testing my online signal processing pipeline. Before I create one from scratch, does anyone have an App to load an XDF file, create the appropriate outlets, then stream the data through the outlets?

I think currently this is only possible using Matlab (to load xdf data). I expect it will be difficult to get the timestamp intervals to match perfectly the intervals in the data file.

@chkothe Do you have XDF import in Python to share that you [mentioned previously](https://github.com/sccn/xdf/issues/3#issuecomment-144538018)? I'd rather write this app in Python than Matlab because I'm more comfortable with Python's multi-processing and I'd like to use a separate process per stream.

**Comment by chkothe** [comment]
Yes, I'm  also not aware of an app that can exactly replicate the timings in the XDF file and that also supports multiple streams. The Python XDF loader isn't yet in any public repo, but I don't see a license issue with it (we could release it under MIT), so I will send it to you for testing. 

By the way, one thing that goes some way towards what you need is the function play_eegset_lsl in BCILAB, which takes in a single-stream+markers EEGLAB dataset struct and which can simulate various types of jitter in the sizes/timings of the blocks and in the reported time stamps. This way you could test the limits of your online signal processing, but it won't scale well to a multi-stream setting. The most recent version is https://github.com/sccn/BCILAB/blob/devel/dependencies/play_eegset_lsl.m and if you go back a few commits you will find a version that does not depend on the argument system in case you want to reduce the dependency footprint (although the version that's included in vis_stream should fit the bill... at some point I really need to put this on the file exchange).

**Comment by cboulay** [comment]
https://github.com/Yida-Lin/XDFStreamer

Yida-Lin is really responsive so even if XDFStreamer doesn't meet my/your needs, this should be a good place to build from.
